datafusion/sql/src/planner.rs

```diff
@@ -103,6 +103,31 @@ pub struct ParserOptions {
     parse_float_as_decimal: bool,
 }
 
+#[derive(Debug, Clone)]
+/// Struct to store Common Table Expression provided with WITH clause and Parameter Data Types provided with PREPARE statement
+pub struct Ctes {
```

Comment:
What do you think about calling this `PlannerState` or `PlannerContext` rather than `Ctes` as will contain more than CTEs?

Reply:
Yes, I will use PlannerContext